### PR TITLE
[improve][broker] improve shadow topic error message

### PR DESCRIPTION
**Pull request by labuladong**
### Motivation

[PIP-180](https://github.com/apache/pulsar/issues/16153) add shadow topic feature. Shadow topic is read-only, but if I send msg to a shadow topic, the error message is `"Only shadow topic supports sending messages with messageId"`, which is confusing.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: 

PTAL @Jason918

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java [hunk 1]
```diff
@@ -210,14 +210,22 @@ public void publishMessage(long producerId, long lowestSequenceId, long highestS
...
                 cnx.getCommandSender().sendSendError(producerId, sequenceId, ServerError.NotAllowedError,
                         "Only shadow topic supports sending messages with messageId");
                 cnx.completedSendOperation(isNonPersistentTopic, headersAndPayload.readableBytes());
             });
             return false;
         }
+        if (isShadowTopic && position == null) {
+            cnx.execute(() -> {
+                cnx.getCommandSender().sendSendError(producerId, sequenceId, ServerError.NotAllowedError,
+                        "Cannot send messages to a shadow topic");
```
Comment (gaozhangmin) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java:224]: We'd better tell this error is caused by `position == null`

Comment (labuladong) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java:224]: This `position` is only used in shadow topic internal replication:

https://github.com/apache/pulsar/blob/5b062f37bd6688c09384abbe908f7c14e28052fc/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L1598-L1602

`position` should be `null` for regular producers. So I think it's not necessary to tell users about this detail.

**Review by Jason918 — APPROVED** [review]
Great, thank you for pointing this out.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java (hunk 1 above)
Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java:224]: > `position` should be `null` for regular producers.

Correct. It's only used in `ShadowReplicator`.

**Comment by labuladong** [comment]
This pr is ready to merge ^_^

**Pull request merged**